### PR TITLE
[ie/mixch] Add comment extraction; update a test

### PR DESCRIPTION
**Pull request by pzhlkj6612**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

info_json:

```JSON
  "comments": [
    {
      "author": "A",
      "author_id": 42,
      "id": "",  // could be an empty string
      "text": "...",
      "timestamp": 1700000000
    },
    {
      "author": "B",
      "author_id": 43,
      "id": "item-1984",
      "text": "...",
      "timestamp": 1700000001
    },

```

.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
> ```json
> 
>       "id": "",  // could be an empty string
> ```

This is not ideal. Is there any other key that could be used as ID?

**Review by pukkandan — APPROVED** [review]

**Comment by pzhlkj6612** [comment]
Hi, @pukkandan .

> This is not ideal. Is there any other key that could be used as ID?

Unfortunately, there is no such key.

So, should we remove "id"?

File: yt_dlp/extractor/mixch.py [hunk 1]
```diff
@@ -48,8 +57,20 @@ def _real_extract(self, url):
...
 
+    def _get_comments(self, video_id):
+        yield from traverse_obj(self._download_json(
+            f'https://mixch.tv/api-web/lives/{video_id}/messages', video_id,
+            note='Downloading comments', errnote='Failed to download comments'), (..., {
+                'author': ('name', {str_or_none}),
+                'author_id': ('user_id', {int_or_none}),
+                'id': ('message_id', {str_or_none}),
+                'text': ('body', {str_or_none}),
+                'timestamp': ('created', {int_or_none}),
```
Comment (pukkandan) [inline yt_dlp/extractor/mixch.py:71]: * It is better to have `id=None` than `id=""`.
* `author_id` should be returned as as a `str`, not `int`.
* `{TYPE_or_none}` converts the data into `TYPE` while `{TYPE}` verifies the data is `TYPE` without converting

Considering these, I believe this is the proper impl:
```suggestion
                'author': ('name', {str}),
                'author_id': ('user_id', {str_or_none}),
                'id': ('message_id', {str}, {lambda x: x or None),
                'text': ('body', {str}),
                'timestamp': ('created', {int}),
```
Correct?

Comment (pzhlkj6612) [reply yt_dlp/extractor/mixch.py]: Thanks! I must have been confused "{TYPE_or_none}" and "{TYPE}" at that time.

Fixed in 5bbd6c18b8a01e2f8966e8f4d586b6ffd4f689ae.

**Review by bashonly — APPROVED** [review]

**Pull request merged**